sdks/java/core/src/test/java/org/apache/beam/sdk/testing/TestPipelineTest.java

```diff
@@ -315,17 +271,28 @@ public void testNoTestPipelineUsedNoAnnotation() {}
     @RunWith(JUnit4.class)
     public static class WithCrashingPipelineRunner {
 
-      static {
-        System.setProperty(TestPipeline.PROPERTY_USE_DEFAULT_DUMMY_RUNNER, Boolean.TRUE.toString());
-      }
-
       private final transient ExpectedException exception = ExpectedException.none();
 
       private final transient TestPipeline pipeline = TestPipeline.create();
 
       @Rule
       public final transient RuleChain chain = RuleChain.outerRule(exception).around(pipeline);
 
+      private static Object oldUseDefaultDummyRunner;
+
+      @BeforeClass
+      public static void beforeClass() {
+        oldUseDefaultDummyRunner =
```

Comment:
Please use the JUnit test rule RestoreSystemProperties instead of a specialized implementation.